gcloud/logging/handlers/transports/background_thread.py

```diff
@@ -0,0 +1,164 @@
+
+                if len(self.batch.entries) > 0:
+                    self.batch.commit()
+        finally:
+            self._entries_condition.release()
+
+        # main thread may be waiting for worker thread to finish writing its
+        # final entries. here we signal that it's done.
+        self._stop_condition.acquire()
+        self._stop_condition.notify()
+        self._stop_condition.release()
+
+    def _start(self):
+        """_start is called by this class's constructor, and is responsible
+        for starting the thread and registering the exit handlers. """
+        try:
+            self._entries_condition.acquire()
+            self._thread = threading.Thread(
+                target=self._run,
+                name="gcloud.logging.handlers.transport.Worker")
+            self._thread.setDaemon(True)
+            self._thread.start()
+        finally:
+            self._entries_condition.release()
+            atexit.register(self._stop)
+
+    def _stop(self):
+        """_stop signals the worker thread to shut down, and waits for
+        stop_timeout second for it to finish.
+
+         _stop is called by the atexit handler registered by
+         :meth:`start`. """
+        if not self.started or self.stopping:
+            return
+
+        # lock the stop condition first so that the worker
+        # thread can't notify it's finished before we wait
+        self._stop_condition.acquire()
+
+        # now notify the worker thread to shutdown
+        self._entries_condition.acquire()
+        self.stopping = True
+        self._entries_condition.notify()
+        self._entries_condition.release()
+
+        # now wait for it to signal it's finished
+        self._stop_condition.wait(self._stop_timeout)
+        self._stop_condition.release()
+        self.stopped = True
+
+    def enqueue(self, record, message):
+        """enqueue queues up a log entry to be written by the background
+        thread. """
+        try:
+            self._entries_condition.acquire()
+            if self.stopping:
+                return
+            self.batch.log_struct({"message": message,
+                                   "python_logger": record.name},
+                                  severity=record.levelname)
+            self._entries_condition.notify()
+        finally:
+            self._entries_condition.release()
+
+
+class BackgroundThreadTransport(Transport):
+    """Aysnchronous tranpsort that uses a background thread to write logging
+    entries as a batch process"""
+
+    def __init__(self, client, name):
+        super(BackgroundThreadTransport, self).__init__(client, name)
+        http = copy.deepcopy(client.connection.http)
+        http = client.connection.credentials.authorize(http)
+        self.client = Client(client.project,
+                             client.connection.credentials,
+                             http)
+        logger = self.client.logger(name)
+        self.worker = _Worker(logger)
+
```

Comment:
@waprin I've seen that traceback before, deep copying the crypto objects associated with a credentials object is not allowed (I think PyCrypto, but maybe one of the other options?). We should work on getting a safe `copy` built into `oauth2client`. I'm happy to give it a try?